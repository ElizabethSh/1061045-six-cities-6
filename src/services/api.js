import axios from 'axios';

const TIMEOUT = 5000;
const BASE_URL = `https://6.react.pages.academy/six-cities`;

const statusCode = {
  UNAUTORIZED: 401,
};

export const createAPI = (unAuthorized) => {
  const api = axios.create({
    baseURL: BASE_URL,
    tiomeout: TIMEOUT,
    withCredentials: true,
  });

  const onSuccess = (response) => response;

  const onError = (error) => {
    const {response} = error;

    if (response.status === statusCode.UNAUTORIZED) {
      unAuthorized();

      throw error;
    }

    throw error;
  };

  axios.interceptors.response.use(onSuccess, onError);

  return api;
};

