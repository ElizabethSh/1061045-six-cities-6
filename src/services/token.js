const LOCALSTORAGE_AUTH_TOKEN = "6-sities-auth-token";

const isLocalStorageAvailable = () =>
  typeof window !== "undefined" && !!window.localStorage;

export const getToken = () => {
  try {
    if (
      isLocalStorageAvailable() &&
      typeof window.localStorage.getItem === "function"
    ) {
      const token = localStorage.getItem(LOCALSTORAGE_AUTH_TOKEN);
      return token ? JSON.parse(token) : undefined;
    }
    return undefined;
  } catch (e) {
    return undefined;
  }
};

export const setToken = (token) => {
  try {
    if (
      isLocalStorageAvailable() &&
      typeof window.localStorage.setItem === "function"
    ) {
      localStorage.setItem(LOCALSTORAGE_AUTH_TOKEN, JSON.stringify(token));
    }
    // eslint-disable-next-line no-empty
  } catch (e) {}
};

export const dropToken = () => {
  try {
    if (
      isLocalStorageAvailable() &&
      typeof window.localStorage.removeItem === "function"
    ) {
      localStorage.removeItem(LOCALSTORAGE_AUTH_TOKEN);
    }
    // eslint-disable-next-line no-empty
  } catch (e) {}
};
