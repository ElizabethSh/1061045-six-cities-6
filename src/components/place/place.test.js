import React from "react";
import { render, screen } from "@testing-library/react";
import { MemoryRouter } from "react-router";
import { Provider } from "react-redux";
import thunk from "redux-thunk";
import configureStore from "redux-mock-store";
import { createAPI } from "../../services/api";
import Place from "./place";
import TestWrapper from "../../services/test-wrapper/test-wrapper";

const api = createAPI(() => {});
const mockStore = configureStore([thunk.withExtraArgument(api)]);

describe(`Test 'Place'`, () => {
  it(`Place should render Loader if data is loading`, () => {
    const store = mockStore({
      PLACE_INFO: { isPlaceInfoLoaded: false },
      NEAR_PLACE: { isNearPlacesLoaded: false },
    });

    render(
      <Provider store={store}>
        <MemoryRouter initialEntries={["/offer/11"]}>
          <Place />
        </MemoryRouter>
      </Provider>
    );

    expect(screen.getByText(/Loading.../i)).toBeInTheDocument();
  });

  it(`Place should render correctly`, () => {
    render(
      <TestWrapper url="/offer/11">
        <Place />
      </TestWrapper>
    );

    expect(screen.getByAltText(/6 cities logo/i)).toBeInTheDocument();
    expect(screen.getByText(/What's inside/i)).toBeInTheDocument();
    expect(screen.getByText(/Meet the host/i)).toBeInTheDocument();
    expect(screen.getByText(/Reviews /i)).toBeInTheDocument();
    expect(
      screen.getByText(/Other places in the neighbourhood/i)
    ).toBeInTheDocument();
  });
});
