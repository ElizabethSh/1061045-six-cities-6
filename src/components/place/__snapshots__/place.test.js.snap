// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Test 'Place' Place should render Loader if data is loading 1`] = `
<div>
  <div
    class="loader"
  >
    <div
      class="loader__dual-ring"
    />
    <p>
      Loading...
    </p>
  </div>
</div>
`;
